### PR TITLE
Add missing boolean combinators

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
# Pull Request Template

## Description

Add missing boolean combinators - like here https://mobily.github.io/ts-belt/api/boolean/
 - `nor`
 - `nand`
 - `xor`
 - `xnor`
 - `implies`

Add XorSemigroup - with name `Semigroup.booleanExclusiveAny` - use it in `xor` combinator

## Types of Changes

- [ ] Bugfix
- [x] New feature
- [ ] Documentation update

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/fp-ts/core/blob/main/.github/CONTRIBUTING.md) for this repository
- [x] I have written tests for the changes I have made
- [x] I have run `pnpm coverage` and my code is 100% covered
- [x] I have updated the documentation (if applicable)
- [x] I agree to license my contribution under the MIT license

## Additional Information

Is there anything else you would like to add? Are there any questions that need to be answered before merging this pull request?

### CODE REVIEW

File: src/Boolean.ts [hunk 1]
```diff
@@ -91,17 +92,33 @@ export const SemigroupAll: semigroup.Semigroup<boolean> = semigroup.booleanAll
...
+ *
+ * assert.deepStrictEqual(SemigroupExclusiveAny.combine(true, true), false)
+ * assert.deepStrictEqual(SemigroupExclusiveAny.combine(true, false), true)
+ * assert.deepStrictEqual(SemigroupExclusiveAny.combine(false, true), true)
+ * assert.deepStrictEqual(SemigroupExclusiveAny.combine(false, false), false)
+ *
+ * @category instances
+ * @since 1.0.0
+ */
+export const SemigroupExclusiveAny: semigroup.Semigroup<boolean> = semigroup.booleanExclusiveAny
```
Comment (gcanti) [inline src/Boolean.ts:120]: What do you think of `SemigroupXor`? It seems shorter and more easily recognizable

Comment (KhraksMamtsov) [reply src/Boolean.ts:120]: Agree - it will be better. I will change

File: src/typeclass/Semigroup.ts [hunk 2]
```diff
@@ -151,6 +151,28 @@ export const booleanAny: Semigroup<boolean> = make(
...
 
+/**
+ * `boolean` semigroup under exclusive disjunction.
+ *
+ * @category instances
+ * @since 1.0.0
+ */
+export const booleanXor: Semigroup<boolean> = make(
+  (self, that) => self !== that,
+  (self, collection) => {
```
Comment (gcanti) [inline src/typeclass/Semigroup.ts:162]: Is the implementation of `combineMany` correct? The following test should pass

```ts
deepStrictEqual(Boolean.SemigroupXor.combineMany(true, [false, true]), false)
```

but with the current implementation fails. 

I'm not sure `combineMany` can be optimised, I would expect a definition like

```ts
export const booleanXor: Semigroup<boolean> = make((self, that) => self !== that)
```

**Comment by jessekelly881** [comment]
Can we add a version of these for Predicates as well?

File: src/typeclass/Semigroup.ts (hunk 2 above)
Comment (KhraksMamtsov) [reply src/typeclass/Semigroup.ts:162]: Agreed, I misunderstood the semantic of `combineMany`. Now it's clear. I will fix it.

File: src/typeclass/Semigroup.ts [hunk 3]
```diff
@@ -151,6 +151,28 @@ export const booleanAny: Semigroup<boolean> = make(
   }
 )
 
+/**
+ * `boolean` semigroup under exclusive disjunction.
+ *
+ * @category instances
+ * @since 1.0.0
+ */
+export const booleanXor: Semigroup<boolean> = make(
```
[the remark this answers is not in the document]
Comment (gcanti) [reply src/typeclass/Semigroup.ts]: `nand` doesn't induces a semigroup instance:

```ts
nand(true, nand(true, false)) = false

nand(nand(true, true), false) = true
```

**Comment by gcanti** [comment]
@KhraksMamtsov aside from the conflict, it looks good to me.

**Review by gcanti — APPROVED** [review]

**Comment by KhraksMamtsov** [comment]
@gcanti in final commit:
- rename `xnor` to `eqv` - is it ok?
- add `MonoidXor` and `MonoidEqv` with tests

Please look at this changes)

**Comment by gcanti** [comment]
> rename xnor to eqv - is it ok?

ok

Note: looks like in `Boolean.ts` is missing `SemigroupEqv`

**Comment by KhraksMamtsov** [comment]
@gcanti done

**Pull request merged**